chalk-integration/src/query.rs

```diff
@@ -132,12 +132,9 @@ fn orphan_check(db: &dyn LoweringDatabase) -> Result<(), ChalkError> {
 
     tls::set_current_program(&program, || -> Result<(), ChalkError> {
         let local_impls = program.local_impl_ids();
+        let mut solver = db.solver_choice().into_solver();
```

Comment:
So to be semantically the same, this should be *inside* the loop (and then you could just pass the solver by value). But I think we only generate a `LocalImplAllowed` goal with no placeholders, so this should be fine to reuse.

Reply:
I've moved it inside the loop for consistency.